src/v/kafka/server/protocol_utils.cc

```diff
@@ -37,8 +38,11 @@ parse_header(ss::input_stream<char>& src) {
     header.key = api_key(reader.read_int16());
     header.version = api_version(reader.read_int16());
     header.correlation = correlation_id(reader.read_int32());
-    auto client_id_size = reader.read_int16();
 
+    // There is a contradiction here with the proposed flexible header
```

Comment:
would it be reasonable to point at a git commit in some popular client implementation that shows this case? mentioning that the KIP is out-of-date from the real world implementation makes sense.

Reply:
i bet Travis B. could comment from the top of his head a few clients. :D

Reply:
```
kmsg: fix AppendRequest for flexible versions

Noticed a regression, asked about it, was pointed to same fix in Kafka
itself:
https://github.com/apache/kafka/pull/7479
```
https://github.com/twmb/franz-go/commit/141969a8

Reply:
haha, told you :D - awesome.